azurerm/internal/services/redisenterprise/redis_enterprise_database_resource.go

```diff
@@ -0,0 +1,458 @@
+package redisenterprise
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/redisenterprise/mgmt/2021-03-01/redisenterprise"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/redisenterprise/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/redisenterprise/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceRedisEnterpriseDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceRedisEnterpriseDatabaseCreate,
+		Read:   resourceRedisEnterpriseDatabaseRead,
+		// Update currently is not implemented, will be for GA
+		// Update: resourceRedisEnterpriseDatabaseUpdate,
+		Delete: resourceRedisEnterpriseDatabaseDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.RedisEnterpriseDatabaseID(id)
+			return err
+		}),
+
+		// Since update is not currently supported all attribute have to be marked as FORCE NEW
+		// until support for Update comes online in the near future
```

Comment:
could we remove the update function if this is the case?

Reply:
Yep. Fixed.